tardis/energy_input/base.py

```diff
@@ -0,0 +1,369 @@
+import numpy as np
+import copy
+from tqdm.auto import tqdm
+import pandas as pd
+
+# from tardis.montecarlo.montecarlo_numba.r_packet import get_random_mu
+from tardis.energy_input.util import SphericalVector, GXPacket, GXPacketStatus
+from tardis.energy_input.gamma_ray_grid import (
+    distance_trace,
+    move_gamma_ray,
+    get_shell,
+    compute_required_packets_per_shell,
+)
+from tardis.energy_input.energy_source import (
+    setup_input_energy,
+    sample_energy_distribution,
+    intensity_ratio,
+)
+from tardis.energy_input.calculate_opacity import (
+    compton_opacity_calculation,
+    photoabsorption_opacity_calculation,
+    pair_creation_opacity_calculation,
+)
+from tardis.energy_input.gamma_ray_interactions import (
+    scatter_type,
+    compton_scatter,
+    pair_creation,
+    get_compton_angle,
+)
+from tardis.energy_input.util import (
+    get_random_theta_gamma_ray,
+    get_random_phi_gamma_ray,
+)
+from tardis import constants as const
+from astropy.coordinates import cartesian_to_spherical
+from tardis.montecarlo.montecarlo_numba.numba_config import CLOSE_LINE_THRESHOLD
+
+
+def initialize_packets(
+    number_of_shells,
+    packets_per_shell,
+    ejecta_volume,
+    inner_velocities,
+    outer_velocities,
+    decay_rad_db,
+):
+    """Initializes packet properties
+    and appends beta decay energy to output tables
+
+    Parameters
+    ----------
+    number_of_shells : int64
+        Number of shells in model
+    packets_per_shell : pandas dataframe
+        Number of packets in a shell
+    ejecta_volume : array of float64
+        Volume per shell
+    inner_velocities : array of float64
+        Shell inner velocities
+    outer_velocities : array of float64
+        Shell outer velocities
+    decay_rad_db : pandas dataframe
+        Decay radiation database
+
+    Returns
+    -------
+    list
+        GXPacket objects
+    numpy array
+        energy binned per shell
+    list
+        packet info
+    """
+    packets = []
+    energy_df_rows = np.zeros(number_of_shells)
+    energy_plot_df_rows = []
+
+    for column in packets_per_shell:
+        subtable = decay_rad_db.loc[column]
+        gamma_ratio, positron_ratio = intensity_ratio(
+            subtable, "'gamma_rays' or type=='x_rays'", "'e+'"
+        )
+        energy_sorted, energy_cdf = setup_input_energy(
+            subtable, "'gamma_rays' or type=='x_rays'"
+        )
+        positron_energy_sorted, positron_energy_cdf = setup_input_energy(
+            subtable, "'e+'"
+        )
+        for shell in range(number_of_shells):
+            required_packets_in_shell = packets_per_shell[column].iloc[shell]
+            for i_packet in range(required_packets_in_shell):
+                # draw a random gamma-ray in shell
+                ray1 = GXPacket(0, 0, 1, GXPacketStatus.IN_PROCESS, 0)
```

Comment:
`ray1` maybe should be `primary_photon`

Reply:
Having named arguments would be useful here. What `0, 0, 1, GXPacketStatus.IN_PROCESS, 0` means is not immediately clear.